hydradx/tests/test_omnipool_amm.py

```diff
@@ -1481,83 +1482,120 @@ def test_oracle_one_block_with_swaps(liquidity: list[float], lrna: list[float],
     for tkn in ['HDX', 'USD', 'DOT']:
         expected_liquidity = omnipool_oracle_0.liquidity[tkn] * (1 - alpha) + alpha * init_liquidity[tkn]['liquidity']
-        if omnipool_oracle_1.liquidity[tkn] != pytest.approx(expected_liquidity, 1e-10):
+        if omnipool_oracle_1.liquidity[tkn] != pytest.approx(expected_liquidity, 1e-12):
             raise AssertionError('Liquidity is not correct.')
 
         expected_vol_in = omnipool_oracle_0.volume_in[tkn] * (1 - alpha) + alpha * vol_in[tkn]
-        if omnipool_oracle_1.volume_in[tkn] != pytest.approx(expected_vol_in, 1e-9):
+        if omnipool_oracle_1.volume_in[tkn] != pytest.approx(expected_vol_in, 1e-12):
             raise AssertionError('Volume is not correct.')
 
         expected_vol_out = omnipool_oracle_0.volume_out[tkn] * (1 - alpha) + alpha * vol_out[tkn]
-        if omnipool_oracle_1.volume_out[tkn] != pytest.approx(expected_vol_out, 1e-9):
+        if omnipool_oracle_1.volume_out[tkn] != pytest.approx(expected_vol_out, 1e-12):
+            raise AssertionError('Volume out is not correct.')
+
+        expected_price = omnipool_oracle_0.price[tkn] * (1 - alpha) + alpha * omnipool_1.lrna_price(tkn)
+        if omnipool_oracle_1.price[tkn] != pytest.approx(expected_price, 1e-12):
+            raise AssertionError('Price is not correct.')
+
+
+def test_oracle_multi_block():
+    init_liquidity = {
+        'HDX': {'liquidity': 100000, 'LRNA': 100000},
+        'USD': {'liquidity': 100000, 'LRNA': 100000},
+        'DOT': {'liquidity': 100000, 'LRNA': 100000},
+    }
+
+    init_oracle = {
+        'liquidity': {'HDX': 100000, 'USD': 100000, 'DOT': 100000},
+        'volume_in': {'HDX': 0, 'USD': 0, 'DOT': 0},
+        'volume_out': {'HDX': 0, 'USD': 0, 'DOT': 0},
+        'price': {'HDX': 1.0, 'USD': 1.0, 'DOT': 1.0},
+    }
+
+    initial_omnipool = oamm.OmnipoolState(
+        tokens=copy.deepcopy(init_liquidity),
+        oracles={
+            'price': 10
+        },
+        asset_fee=0.0025,
+        lrna_fee=0.0005,
+        last_oracle_values={
+            'price': copy.deepcopy(init_oracle)
+        }
+    )
+
+    initial_state = GlobalState(
+        pools={'omnipool': initial_omnipool},
+        agents={'trader': Agent(
+            enforce_holdings=False,
+            trade_strategy=schedule_swaps(
+                'omnipool', swaps=[
+                    *[None] * 4,
+                    {'tkn_sell': 'DOT', 'tkn_buy': 'HDX', 'sell_quantity': 1000},
+                    *[None] * 4,
+                    {'tkn_sell': 'HDX', 'tkn_buy': 'DOT', 'buy_quantity': 1000},
+                ]
+            )
+        )}
+    )
+
+    events = run.run(initial_state=initial_state, time_steps=10, silent=True)
+    final_omnipool = events[-1].pools['omnipool'].update()
+    omnipool_oracle = final_omnipool.oracles['price']
+    expected_vol_in = {'DOT': 232.21719930388855, 'HDX': 622.8414188596074, 'USD': 0}
```

Comment:
I know this is harder to set up, but I think it'd be much better to test against our old setup of updating oracles every block, rather than test against hardcoded values? We should test the 1-block oracle update in other ways (potentially against hardcoded values), but those tests should already be in place I think?

Reply:
That's what this one does:
test_update_every_block